daemon/src/maker_cfd.rs

```diff
@@ -64,8 +64,8 @@ pub struct Actor {
     current_order_id: Option<OrderId>,
     monitor_actor: Address<monitor::Actor<Actor>>,
     setup_state: SetupState,
-    latest_announcement: Option<oracle::Announcement>,
-    _oracle_actor: Address<oracle::Actor<Actor, monitor::Actor<Actor>>>,
+    latest_announcements: Option<oracle::Announcements>,
```

Comment:
You can use a `BTreeMap` here, that is a sorted `HashMap` so you can grab the last one but still look up by ID.